app/models/grda_warehouse/client_matcher_lookups/proper_name_lookup.rb

```diff
@@ -36,6 +38,9 @@ def add(client)
     def normalize(str)
       return nil unless str.present?
 
+      # Convert to UTF-8 as the regex expects it
+      str = str.encode('UTF-8', 'ASCII-8BIT', invalid: :replace, undef: :replace, replace: '')
```

Comment:
This is forcing the source to be treated as ascii even if it is utf8. I might be missing something but it looks like this would strip out valid utf8 chars:
```
> utf8_str = "José"
=> "José"
> result = utf8_str.encode("UTF-8", "ASCII-8BIT", invalid: :replace, undef: :replace, replace: '')
=> "Jos"
```

So maybe we should make this conditional. Assuming that str.valid_encoding can detect this issue. Also maybe we should replace with '?' so that users can see that there was an encoding problem and the data is incomplete
```suggestion
      str = str.encode('UTF-8', 'ASCII-8BIT', invalid: :replace, undef: :replace, replace: '?') unless str.valid_encoding?
```

Reply:
Thanks!
Looks like if we don't set `transliterate = true` it throws out non-ascii as well.  Not sure why we aren't transliterating for Identify Duplicates, but for now, I'm not changing it.  
I didn't add `?` as these are never seen, just used in automated matching routines.

I've added a check to only re-encode if the source is ASCII-8BIT, and some tests to expose the current functionality.  Let me know if you think it should be handled differently.

Reply:
got it, that should be fine. If we trust str.encoding we could probably leave off the condition and let it auto detect the source too, I guess that would handle other encodings potentially
```
str = str.encode('UTF-8',  invalid: :replace, undef: :replace, replace: '')
```

Reply:
I'm up for removing the source encoding, I've adjusted the condition to only run if it isn't already UTF-8.